### PR TITLE
Release candidate

### PR DESCRIPTION
**Pull request by krl**

### CODE REVIEW

File: src/compound.rs [hunk 1]
```diff
@@ -4,183 +4,115 @@
...
-    A: Annotation<Self::Leaf>,
-{
+pub trait Compound<A, S>: Sized + Archive {
     /// The leaf type of the Compound collection
-    type Leaf;
+    type Leaf: Archive;
 
     /// Get a reference to a child    
-    fn child(&self, ofs: usize) -> Child<Self, A>;
+    fn child(&self, ofs: usize) -> Child<Self, A, S>
```
Comment (miloszm) [inline src/compound.rs:78]: It was not added in this PR, so what follows is only a suggestion/comment - I think it'd be simpler if child method returned a Compound type, and what is now Child enum would be treated as "this node" enum. This way we could eliminate the Leaf type variable. It is confusing that not every node is a Compound, and that there is a difference between "this" which is a Compound, and a child, which is a Child. I'd make everything a Compound and rename Child to Node and keep it as sth like "current" or "this" inside a compound (or merge Child (Node) with Compound).

**Review by miloszm — CHANGES_REQUESTED** [review]
LGTM, asking for minor changes

File: src/link.rs [hunk 2]
```diff
@@ -209,8 +230,10 @@ impl<C, A> Link<C, A> {
                 return Rc::make_mut(rc);
             }
             Link::Archived { stored, .. } => {
-                let ca = Portal::get(*stored);
-                let c = ca.deserialize(&mut Infallible).expect("Infallible");
+                let mut store = stored.store().clone();
```
[the remark this answers is not in the document]
Comment (krl) [reply src/link.rs:233]: renamed to Link::Stored

File: src/link.rs [hunk 3]
```diff
@@ -67,56 +79,61 @@ impl<C, A> Archive for Link<C, A> {
...
                     a
                 };
                 let to_insert = &(**rc);
-                let ofs = ser.borrow_mut().put(to_insert);
-                Ok((ofs, a))
+                let ident = Ident::new(ser.put(to_insert));
+                Ok((ident, a))
+            }
+            Link::Archived { .. } => {
+                unreachable!("FIXME motivate why this does not happen")
```
[the remark this answers is not in the document]
Comment (krl) [reply src/link.rs:127]: Fixed, and provded blanket impl since all fields for the resolver are available

File: src/compound.rs (hunk 1 above)
Comment (krl) [reply src/compound.rs:78]: i'm not sure if i understand what you mean here.

**Review by miloszm — APPROVED** [review]
LGTM

**Pull request merged**